pkg/api/types.go

```diff
@@ -578,6 +578,7 @@ type AgentPoolProfile struct {
 	AvailabilityProfile                 string               `json:"availabilityProfile"`
 	ScaleSetPriority                    string               `json:"scaleSetPriority,omitempty"`
 	ScaleSetEvictionPolicy              string               `json:"scaleSetEvictionPolicy,omitempty"`
+	SpotMaxPrice                        *float64             `json:"spotMaxPrice,omitempty"`
```

Comment:
let's also add the type to vlabs so we can test it in aks-engine E2E

Reply:
that will require changing pkg/api/vlabs/types.go, pkg/api/vlabs/types_test.go (copy the same test as the one in api/types.go) pkg/api/convertertoapi.go and pkg/api/converterfromapi.go

Reply:
Thanks Cecile! Added field to vlabs and related tests. Also added into examples/kubernetes-vmss-spot